app/forms/instance-create.tsx

```diff
@@ -328,17 +343,7 @@ export function CreateInstanceForm() {
         label="Disk size"
         name="bootDiskSize"
         control={control}
-        // Imitate API logic: only require that the disk is big enough to fit the image
-        validate={(diskSizeGiB) => {
-          if (!image) return true
-          if (diskSizeGiB < image.size / GiB) {
-            const minSize = Math.ceil(image.size / GiB)
-            return `Must be as large as selected image (min. ${minSize} GiB)`
-          }
-          if (diskSizeGiB > MAX_DISK_SIZE_GiB) {
-            return `Can be at most ${MAX_DISK_SIZE_GiB} GiB`
-          }
-        }}
+        imageSize={image?.size ? Math.ceil(image.size / GiB) : undefined}
```

Comment:
I'm not quite sure that we should should push the validation logic down into the `DiskSizeField` component. Disks, as a concept, aren't strictly speaking tied to images. Certainly in this case the disk size is semi-dependent on the image size, but that doesn't impact other situations in which a disk size may be specified. 

We'll either continue pushing more context down into the `DiskSizeField` component or have to find some other way of composing the logic. I think my preference, for now, is to localize the validation logic to where it's used (similar to what we had here). If we need that logic to be more portable perhaps we could just have some reusable helper we could spread into the usage?

Reply:
Yeah, even though the logic is currently the same between the two spots, we could model it as a shared `validate` function that’s passed in in both spots.

Reply:
Refactored this after chatting with @david-crespo … several of the validations are standard and will be needed for the DiskSizeField regardless of where we use it. But we also want to have the option of adding one-off validators. So we added an optional `validate` prop that we can pass in when calling the component; any validations passed in via that prop will get called after the standard validator functions are evaluated.